src/custom/state/sentry/updater/index.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect } from 'react'
+import * as Sentry from '@sentry/browser'
+
+import useIsWindowVisible from 'hooks/useIsWindowVisible'
+import { useSwapState } from 'state/swap/hooks'
+import { SupportedChainId } from 'constants/chains'
+import { useCurrency } from 'hooks/Tokens'
+import { useWalletInfo } from 'hooks/useWalletInfo'
+
+export default function Updater(): null {
+  const { account, chainId, walletName, isSupportedWallet } = useWalletInfo()
+  const windowVisible = useIsWindowVisible()
+
+  const {
+    INPUT: { currencyId: sellTokenAddress },
+    OUTPUT: { currencyId: buyTokenAddress },
+  } = useSwapState()
+
+  const sellCurrency = useCurrency(sellTokenAddress)
+  const buyCurrency = useCurrency(buyTokenAddress)
+
+  // create sentry context based on "main" parameters
+  useEffect(() => {
+    if (windowVisible) {
+      Sentry.setContext('user', {
+        userAddress: account || 'DISCONNECTED',
+        wallet: walletName,
+        isWalletSupported: isSupportedWallet,
+        network:
```

Comment:
This network part is a bit hard to read and also if the selected network is unsupported it will return Mainnet.

Reply:
Good point @nenadV91 !

Reply:
Lets use chainId if we cannot resolve the name. I agree on this being too hard to read